specification.md

```diff
@@ -106,51 +106,187 @@ incoming events can trigger function invocations during workflow execution.
+The following sections provide detailed descriptions of the Serverless Workflow model. 
+
+For each model definition section we provide:
 
 - Parameter description in table format.
-- Example definition in both [JSON](https://www.json.org/json-en.html) and [YAML](https://yaml.org/) formats.
+- Example definition in both [JSON](https://www.json.org/json-en.html) and [YAML](https://yaml.org/) formats.\
+- Detailed definition explanation with addition examples where needed.
 
 ### Workflow Model
 
-Serverless Workflow model can be viewed as: 
+The Serverless Workflow model can be viewed as: 
+
 * Set of [functions](#Function-Definition) (services) that need to be called during workflow execution. 
-* Set of [events](#Event-Definition) that need to be consumed to start workflow instances or trigger functions, or produced during workflow execution.
-* Set of [states](#State-Definition) and [transitions](#Transitions) between them. States define the control flow logic, manage data, and can reference defined functions and events.
+* Set of [events](#Event-Definition) that need to be consumed to start workflow instances or trigger functions, or be produced during workflow execution.
+* Set of [states](#State-Definition) and [transitions](#Transitions) between them. States define the workflow control flow logic, manage data, and can reference defined functions and events.
+
+The defined workflow model is a declarative language that can be used to model small or complex orchestrations
+for event-driven, serverless applications.
+
+### Workflow Data
+
+Serverless Workflow data is represented in [JSON](https://www.json.org/json-en.html) format.
+Data flow during workflow execution can be divided into:
+
+- [Workfow data input](#Workflow-data-input)
+- [Event data](#Event-data)
+- [Action data](#Action-data)
+- [Information passing between states](#Information-passing-Between-States)
+- [State information filtering](#State-information-filtering)
+- [Workflow data output](#Workflow-data-output)
 
-The model defines a declarative language that can be used to define small or complex workflow orchestrations.
+### Workflow Data Expressions
 
-### Workflow Expressions
+Workflow model parameters may use expressions to access the JSON data. 
+Note that different data filters play a big role as to which parts of the state data is queried against. Reference the 
+[State Information Filtering](#State-Information-Filtering) section for more information.
 
-Different parts of the workflow definition such as [transitions](#Transition-Definition), [retries](#Retry-Definition), [errors](#Error-Definition), 
-etc include "expression" parameters. 
-These parameters define expressions that have to be evaluated against state or error data.
+All expressions must follow the [JsonPath](https://github.com/json-path/JsonPath) format and can be evaluated with a JsonPath expression evaluator. 
+JSONPath expressions can traverse the workflow JSON data. 
```

Comment:
```suggestion
JsonPath expressions can traverse the workflow JSON data. 
```

Reply:
fixed and also updated all places to change from "JSONPath" to "JsonPath"